Execution failed for task ':permission_handler_android:compileDebugJavaWithJavac'.

## 🐛 Bug Report

<!--- Summary description of the bug --->
error: warnings found and -Werror specified
1 error


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':permission_handler_android:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.


### Expected behavior
Should compile

### Reproduction steps
Run flutter run

### Configuration

**Version:** 9.2.0

**Platform:** 
- [ ] :iphone: iOS
- [ X] :robot: Android

Hi, I have the same issue with `permission_handler: ^10.4.0`. It works fine when building the app (I see no errors), but it throws when trying to run integration tests using Patrol.

```bash
	FAILURE: Build failed with an exception.
	
	* What went wrong:
	Execution failed for task ':permission_handler_android:compileDebugJavaWithJavac'.
	> Compilation failed; see the compiler error output for details.
	
	* Try:
	> Run with --stacktrace option to get the stack trace.
	> Run with --info or --debug option to get more log output.
	> Run with --scan to get full insights.
	
	* Get more help at https://help.gradle.org
```

We have these sdk settings: `compileSdkVersion 33`, `minSdkVersion 23`, `targetSdkVersion 33`.

@kashiflab 
Currently, I can successfully compile and build an APK for my Flutter application without encountering any issues. However, as soon as I include the 'permission_handler' package, I am unable to do so. I have tried using multiple versions of the 'permission_handler' package, and compileSdkVersion 33 targetSdkVersion 33 but the problem persists. The error message I receive is as follows:

error: warnings found and -Werror specified

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':permission_handler_android:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

--- a/permission_handler_android/CHANGELOG.md
+++ b/permission_handler_android/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 10.3.3
+
+* Migrates the Gradle compile arguments to the example app, so they are not enforced upon consumers of the plugin.
+
 ## 10.3.2
 
 * Updates example app to show `Permission.photos` and hide `Permission.bluetooth`.

--- a/permission_handler_android/android/build.gradle
+++ b/permission_handler_android/android/build.gradle
@@ -1,6 +1,5 @@
 group 'com.baseflow.permissionhandler'
 version '1.0'
-def args = ["-Xlint:deprecation","-Xlint:unchecked","-Werror"]
 
 buildscript {
     repositories {
@@ -18,10 +17,6 @@ rootProject.allprojects {
         google()
         mavenCentral()
     }
-}
-
-project.getTasks().withType(JavaCompile){
-    options.compilerArgs.addAll(args)
 }
 
 apply plugin: 'com.android.library'

--- a/permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionUtils.java
+++ b/permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionUtils.java
@@ -122,25 +122,24 @@ public class PermissionUtils {
                 break;
 
             case PermissionConstants.PERMISSION_GROUP_LOCATION_ALWAYS:
+            case PermissionConstants.PERMISSION_GROUP_LOCATION_WHEN_IN_USE:
+            case PermissionConstants.PERMISSION_GROUP_LOCATION:
                 // Note that the LOCATION_ALWAYS will deliberately fallthrough to the LOCATION
                 // case on pre Android Q devices. The ACCESS_BACKGROUND_LOCATION permission was only
                 // introduced in Android Q, before it should be treated as the ACCESS_COARSE_LOCATION or
                 // ACCESS_FINE_LOCATION.
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                if (permission == PermissionConstants.PERMISSION_GROUP_LOCATION_ALWAYS && Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                     if (hasPermissionInManifest(context, permissionNames, Manifest.permission.ACCESS_BACKGROUND_LOCATION))
                         permissionNames.add(Manifest.permission.ACCESS_BACKGROUND_LOCATION);
                     break;
                 }
-            case PermissionConstants.PERMISSION_GROUP_LOCATION_WHEN_IN_USE:
-            case PermissionConstants.PERMISSION_GROUP_LOCATION:
+
                 if (hasPermissionInManifest(context, permissionNames, Manifest.permission.ACCESS_COARSE_LOCATION))
                     permissionNames.add(Manifest.permission.ACCESS_COARSE_LOCATION);
 
                 if (hasPermissionInManifest(context, permissionNames, Manifest.permission.ACCESS_FINE_LOCATION))
                     permissionNames.add(Manifest.permission.ACCESS_FINE_LOCATION);
                 break;
-
-
             case PermissionConstants.PERMISSION_GROUP_SPEECH:
             case PermissionConstants.PERMISSION_GROUP_MICROPHONE:
                 if (hasPermissionInManifest(context, permissionNames, Manifest.permission.RECORD_AUDIO))
@@ -191,7 +190,7 @@ public class PermissionUtils {
                         permissionNames.add(Manifest.permission.BODY_SENSORS_BACKGROUND);
                     }
                 }
-
+                break;
             case PermissionConstants.PERMISSION_GROUP_SMS:
                 if (hasPermissionInManifest(context, permissionNames, Manifest.permission.SEND_SMS))
                     permissionNames.add(Manifest.permission.SEND_SMS);

--- a/permission_handler_android/example/android/build.gradle
+++ b/permission_handler_android/example/android/build.gradle
@@ -16,12 +16,6 @@ allprojects {
     }
 }
 
-def args = ["-Xlint:deprecation","-Xlint:unchecked","-Werror"]
-
-project.getTasks().withType(JavaCompile){
-    options.compilerArgs.addAll(args)
-}
-
 rootProject.buildDir = '../build'
 subprojects {
     project.buildDir = "${rootProject.buildDir}/${project.name}"
@@ -30,4 +24,16 @@ subprojects {
 
 tasks.register("clean", Delete) {
     delete rootProject.buildDir
+}
+
+// Build the plugin project with warnings enabled. This is here rather than
+// in the plugin itself to avoid breaking clients that have different
+// warnings (e.g., deprecation warnings from a newer SDK than this project
+// builds with).
+gradle.projectsEvaluated {
+    project(":permission_handler_android") {
+        tasks.withType(JavaCompile) {
+            options.compilerArgs << "-Xlint:all" << "-Werror"
+        }
+    }
 }

--- a/permission_handler_android/example/android/build.gradle
+++ b/permission_handler_android/example/android/build.gradle
@@ -16,6 +16,12 @@ allprojects {
     }
 }
 
+def args = ["-Xlint:deprecation","-Xlint:unchecked","-Werror"]
+
+project.getTasks().withType(JavaCompile){
+    options.compilerArgs.addAll(args)
+}
+
 rootProject.buildDir = '../build'
 subprojects {
     project.buildDir = "${rootProject.buildDir}/${project.name}"

--- a/permission_handler_android/pubspec.yaml
+++ b/permission_handler_android/pubspec.yaml
@@ -1,7 +1,7 @@
 name: permission_handler_android
 description: Permission plugin for Flutter. This plugin provides the Android API to request and check permissions.
 homepage: https://github.com/baseflow/flutter-permission-handler
-version: 10.3.2
+version: 10.3.3
 
 environment:
   sdk: ">=2.15.0 <4.0.0"

